tallycash/extension
# Add Webpack configs with experimental feature set

Builds two variants per supported browser:
the first one has only some features enabled (ready for the public),
the second ("experimental") has all the features enabled (for QA).

The two feature sets are configured respectively with the files
`.env.features` and `.env.features.experimental`.

The build target location is unchanged for the standard feature set,
whereas experimental builds include the suffix  `-experimental`.
For example, for the Chrome browser, we now build:
`chrome/`, `chrome.zip`,
`chrome-experimental/` and `chrome-experimental.zip`.

## What to test

- Check that the build generates all the variants in the expected location.
- Install a standard build and an experimental one, and check that the enabled features change between the two as expected (for one feature at least).
- Make sure that CI is not broken, as well as anything that depends on it.
- Make sure that local environment configuration continues to work, so that we do not break any dev workflow.
- Evaluate the cost/benefit of doubling-up the duration of all builds. (Perhaps we could build experimental variants for `firefox` and `chrome` only?)

## Discussion

Now that we depend on three parameters (mode, browser, features) instead of two, `webpack.config.ts` may benefit from a further reorganization. (Not done in this PR to reduce its code footprint.) For example, all the configuration could be merged into a single function, which first builds a collection of partial configurations (depending on mode, browser and features) and then merges them together at the very last step.
